x/ibc/07-tendermint/types/header.go

```diff
@@ -54,7 +54,7 @@ func (h Header) GetTime() time.Time {
 // that validatorsets are not nil.
 // NOTE: TrustedHeight and TrustedValidators may be empty when creating client
 // with MsgCreateClient
-func (h Header) ValidateBasic(chainID string) error {
```

Comment:
I guess we could just call the header validate basic in the update func for tendermint since this needs to be modular

Reply:
this was renamed to Validate, the check is still done during message validation. Do you want me to readd the chainID param?

Reply:
I don't follow. Where is the check done? Specifically in updates we need to pass in clientState.GetChainID to `tmSignedHeader.ValidateBasic()` and in misbehaviour we need to pass in `misbehaviour.GetChainID` to `header.tmSignedHeader.ValidateBasic`

we cannot assume the headers chainID == client state/misbehaviour. Passing in the chainID to `ValidateBasic` is annoying because then we have to modify the interface for all clients which I'd prefer not to do since validatebasic is supposed to work without context.

I'd suggest we leave as is here, but add checks to misbehaviour and update to specifically check these conditions. The check is [basic](https://github.com/tendermint/tendermint/blob/55cd49ad081c95d79eb5376c1a6c5c8d9af35baa/types/block.go#L1019) so we can just add our own version without calling signedHeader.ValidateBasic again.

I'd like to hear what @AdityaSripal thinks

Reply:
> I don't follow. Where is the check done?

On `MsgUpdateClient.ValidateBasic`. I guess I can leave the interface with the `chainID string` param

Reply:
@colin-axner if we keep the `chainID` field we can't call `header.ValidateBasic` on `MsgUpdateClient.ValidateBasic`. Thoughts?

Reply:
I'd prefer to not add the `chainID` param and do individual checks in tendermint as necessary. This just means adding a check in misbehaviour_handle (misbehaviour.ChainID == header.ChainID && clientState.ChainID == header.ChainID) and update_handle (clientState.ChainID == header.ChainID)

Reply:
See my comment below. We can remove passing in `chainID` here, but we must check this elsewhere in the logic that calls `header.ValidateBasic` as @colin-axner points out

I pointed those places out in my review, should be a very quick fix